### PR TITLE
next: save user profile to record gear

### PR DESCRIPTION
**Pull request by kaojohnny**
connects #755

## note ##
1. Based on #757
2. For record to record communication, handler should commit transaction before the request initiate.

### CODE REVIEW

File: pkg/auth/handler/signup.go [hunk 1]
```diff
@@ -223,20 +262,65 @@ func (h SignupHandler) verifyPayload(payload SignupRequestPayload) (err error) {
...
+	if err = h.AuthInfoStore.DeleteAuth(authInfoID); err != nil {
+		return
+	}
+
+	if err = h.TxContext.CommitTx(); err != nil {
+		return
+	}
+
+	if err != nil {
+		err = h.TxContext.RollbackTx()
```
Comment (Steven-Chan) [inline pkg/auth/handler/signup.go]: Should rollback the transaction for early return if error exists.

File: pkg/auth/dependency/userprofile/tx_provider.go [hunk 2]
```diff
@@ -22,12 +23,17 @@ func NewSafeProvider(
...
 	return s.impl.CreateUserProfile(userID, data)
 }
 
-func (s *safeUserProfileImpl) GetUserProfile(userID string) (profile UserProfile, err error) {
+func (s *safeUserProfileImpl) GetUserProfile(userID string, accessToken string) (profile UserProfile, err error) {
 	s.txContext.EnsureTx()
 	return s.impl.GetUserProfile(userID)
 }
+
+func (s *safeUserProfileImpl) WithInTx() (withIn bool) {
```
Comment (Steven-Chan) [inline pkg/auth/dependency/userprofile/tx_provider.go]: I thought `WithInTx` was saying if the store is now in transaction.

I suggest to use a function name that describe the store characteristics, something like `IsOperationRunInDB` or `CanWithInTx`.

**Comment by kaojohnny** [comment]
@Steven-Chan Thanks. Updated.

**Comment by kaojohnny** [comment]
Auth gear will use record gear API to create records directly (avoid gear to gear request).

**Pull request closed without merging**